File: Python/7py 2/is_nice.py
"""
|--------------------------------------------------------------------------
| Nice Array
|--------------------------------------------------------------------------
|
"""

def is_nice(arr):
   res = 0
   for i in arr:
      if i+1 in arr or i-1 in arr:
         res += 1
   return len(arr) != 0 and res == len(arr)


def is_nice(arr):
   return len(arr) > 0 and all(n-1 in arr or n+1 in arr for n in arr)


def is_nice(arr):
   return bool(a := set(arr)) and next((False for n in a if (n + 1) not in a and (n - 1) not in a), True)


def is_nice(arr):
   s = set(arr)
   return bool(arr) and all( n+1 in s or n-1 in s for n in s)


print(is_nice([2,10,9,3]),True)
print(is_nice([]),False)

File: PHP/7php 2/isNice.php
<?php
/*
|--------------------------------------------------------------------------
| Nice Array
|--------------------------------------------------------------------------
|
*/

function isNice($arr) {
   $res = 0;
   foreach ($arr as $i) {
      if (in_array($i+1, $arr) || in_array($i-1, $arr)) $res += 1;
   }
   return count($arr) != 0 && $res == count($arr);
}


function isNice($arr) {
   foreach($arr as $i) {
      if(!(in_array($i - 1, $arr) || in_array($i + 1, $arr))) 
         return false;
   }
   return $arr ? true : false;
}


function isNice($arr) {
   foreach($arr as $n){
      if (!in_array($n-1,$arr)&&!(in_array($n+1,$arr))) return false;
   }
   return !!count($arr);
}


print_r(isNice([2, 10, 9, 3]) . ' ' . true . '
');
print_r(isNice([]) . ' ' . false . '
');